src/middlewared/middlewared/plugins/init_shutdown_script.py

```diff
@@ -149,9 +169,24 @@ class Config:
             stdout, stderr = await proc.communicate()
 
             if proc.returncode:
+                if task_type == 'COMMAND':
+                    cmd = task['command']
+                elif task_type == 'SCRIPT' and task['script_text']:
+                    cmd = task['script_text']
+                elif task_type == 'SCRIPT' and task['script']:
+                    cmd = task['script']
+                else:
+                    cmd = ''
                 self.middleware.logger.debug(
                     'Execution failed for '
-                    f'{task_type} {task["command"] if task_type == "COMMAND" else task["script"]}: {stderr.decode()}'
+                    f'{task_type} {cmd}: {stderr.decode()}'
+                )
+        if tmp_script:
+            try:
+                os.unlink(tmp_script)
```

Comment:
I think you may have misunderstood me in the earlier comment.

Say an exception occurs anywhere between the creation of the tmpfile and this (e.g. subprocess communicate for w/e reason), what will happen? File will stay there forever?